### PR TITLE
Added default config files feature

### LINKED ISSUE #70: [feature] Create default config files
If the config file isn't found the default config will be used.

Add the capability to write the default config file in the data-dir as: `~/.coinswap/configs/maker.toml` (and `taker.toml`).

### PR DESCRIPTION
**Pull request by Shourya742**
This PR closes #70

### CODE REVIEW

File: src/maker/config.rs [hunk 1]
```diff
@@ -61,22 +61,30 @@ impl MakerConfig {
...
             }
         } else {
             let default_path = PathBuf::from("maker.toml");
             if default_path.exists() {
                 parse_toml(&default_path)?
             } else {
+                let default_maker_config_path = get_config_dir().join("maker.toml");
+                if !default_maker_config_path.exists() {
+                    create_default_maker_dirs(&default_maker_config_path);
+                }
```
Comment (rajarshimaitra) [inline src/maker/config.rs:82]: This logic needs simplification as follow:

- No custom datadir is provided, so config file should be checked at default datadir, `get_config_dir()`.
- If present, read the config file from there.
- If not, create a default config at that location.

No need to check for `maker.toml` in the current repo directory as that's nonstandard. We should keep the `maker.toml` in repo directory just as an example. But operations should always be done at either custom datadir, or at default datadir.

File: src/maker/config.rs [hunk 2]
```diff
@@ -61,22 +61,30 @@ impl MakerConfig {
...
                 log::warn!(
                     "Maker config file not found in default path: {}, using default config",
```
Comment (rajarshimaitra) [inline src/maker/config.rs:84]: Update the warning message acordingly.

File: src/maker/config.rs [hunk 3]
```diff
@@ -143,6 +151,27 @@ impl MakerConfig {
     }
 }
 
+fn create_default_maker_dirs(default_maker_config_path: &PathBuf) {
```
Comment (rajarshimaitra) [inline src/maker/config.rs:154]: A more clearer name for this could be

```suggestion
fn create_default_maker_config(target_path: &PathBuf) {
```

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
Concept ACK.

Few approach fixes below.

File: src/maker/config.rs (hunk 1 above)
Comment (Shourya742) [reply src/maker/config.rs:82]: ok ... Got it

Comment (Shourya742) [reply src/maker/config.rs:82]: We are checking this in case the datadir is None.

**Comment by rajarshimaitra** [comment]
I am not sure if its ready for review. I don't see this comment addressed.. 

https://github.com/utxo-teleport/teleport-transactions/pull/73#discussion_r1433623264

Also it has some conflict..

@Shourya742 if you need hands with it, I can push the fixes.

File: src/maker/config.rs [hunk 4]
```diff
@@ -61,22 +61,27 @@ impl MakerConfig {
...
+                    "Maker config file not found, creating a default file at path : {}, ",
                     path.display()
                 );
-                HashMap::new()
+                parse_toml(&path)?
             }
         } else {
             let default_path = PathBuf::from("maker.toml");
             if default_path.exists() {
                 parse_toml(&default_path)?
```
Comment (rajarshimaitra) [inline src/maker/config.rs:74]: This code should is not needed anymore. There is no need for the config files at repo dir, As they will always be created at runtime in default/custom datadir if doesn't exist. 

We should document this in our readme. As well as in the function doc.

File: src/maker/config.rs [hunk 5]
```diff
@@ -1,8 +1,8 @@
 //! Maker Configuration. Controlling various behaviors.
 
-use std::{collections::HashMap, io, path::PathBuf};
+use std::{io, path::PathBuf};
 
-use crate::utill::{parse_field, parse_toml};
+use crate::utill::{get_config_dir, parse_field, parse_toml, write_default_config};
 
 /// Maker Configuration, controlling various maker behavior.
```
Comment (rajarshimaitra) [inline src/maker/config.rs:7]: The current doc is wrong..

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
```rust
/// new a default configuration with given port and address
pub fn new(file_path: Option<&PathBuf>) -> io::Result<Self> {
```

The current doc is missleading.

File: src/maker/config.rs (hunk 4 above)
Comment (Shourya742) [reply src/maker/config.rs:74]: ok

**Review by rajarshimaitra — APPROVED** [review]
ACK. 

I think I messed up the commit history while trying to fix the conflict and make some improves.

Opening up a new one.

**Comment by rajarshimaitra** [comment]
overridden by #75

**Pull request closed without merging**